yt_dlp/extractor/allstar.py

```diff
@@ -0,0 +1,259 @@
+            'thumbnail': 'md5:60872f0d236863bb9a6f3dff1623403c',
+            'uploader': 'cherokee',
+            'ext': 'mp4',
+            'timestamp': 1688365434271,
+            'uploader_id': '62b8bdfc9021052f7905882d',
+            'view_count': int,
+        }
+    }]
+
+    def _real_extract(self, url):
+        query_id, video_id = self._match_valid_url(url).groups()
+
+        return self._parse_video_data(
+            self._send_query(
+                _QUERIES.get(query_id), {'id': video_id},
+                ('data', 'video'), video_id))
+
+
+class AllstarProfileIE(AllstarBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?allstar\.gg/(?:profile\?user=|u/)(?P<id>[^/?#&]+)'
+
+    _TESTS = [{
+        'url': 'https://allstar.gg/profile?user=62b8bdfc9021052f7905882d',
+        'info_dict': {
+            'id': '62b8bdfc9021052f7905882d',
+            'title': 'cherokee - Clips',
+        },
+        'playlist_mincount': 15
+    }, {
+        'url': 'https://allstar.gg/u/cherokee?game=730&view=Clips',
+        'info_dict': {
+            'id': '62b8bdfc9021052f7905882d',
+            'title': 'cherokee - Clips',
+        },
+        'playlist_mincount': 15
+    }, {
+        'url': 'https://allstar.gg/u/62b8bdfc9021052f7905882d?view=Montages',
+        'info_dict': {
+            'id': '62b8bdfc9021052f7905882d',
+            'title': 'cherokee - Montages',
+        },
+        'playlist_mincount': 4
+    }, {
+        'url': 'https://allstar.gg/profile?user=cherokee&view=Mobile Clips',
+        'info_dict': {
+            'id': '62b8bdfc9021052f7905882d',
+            'title': 'cherokee - Mobile Clips',
+        },
+        'playlist_mincount': 1
+    }]
+
+    _PAGE_SIZE = 10
+
+    @staticmethod
+    def _set_webpage_url(info_dict):
+        video_id = info_dict.get('id')
+        video_url = info_dict.get('url')
+
+        if video_url is None or video_id is None:
+            return info_dict
+
+        base_name = 'clip' if '/clips/' in video_url else 'montage'
+        info_dict['webpage_url'] = f'https://allstar.gg/{base_name}?{base_name}={video_id}'
+        info_dict['webpage_url_basename'] = base_name
+
+        return info_dict
+
+    def _get_page(self, user_id, display_id, game, query, page_num):
+        page_num += 1
+
+        for video_data in self._send_query(
+                query, {
+                    'user': user_id,
+                    'page': page_num,
+                    'game': game,
+                }, ('data', 'videos', 'data'), display_id, f'Downloading page {page_num}'):
+            yield self._set_webpage_url(self._parse_video_data(video_data))
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        profile_data = self._download_json(
+                urljoin('https://api.allstar.gg/v1/users/profile/', display_id),
+                display_id)
+        user_id = traverse_obj(profile_data, ('data', ('_id'), {str_or_none}))
```

Comment:
```suggestion
        user_id = traverse_obj(profile_data, ('data', ('_id'), {str}))
```

Reply:
if `_id` is given as an int value by the site, we do want to use `str_or_none`

Reply:
Since the site uses hexadecimal ids they are given as a string.